test/Kestrel.Transport.FunctionalTests/ResponseTests.cs

```diff
@@ -802,7 +803,14 @@ async Task App(HttpContext context)
             Assert.False(requestAborted);
         }
 
+        private bool ConnectionNotClosedWhenClientSatisfiesMinimumDataRateGivenLargeResponseHeadersRetryPredicate(Exception e)
+            => e is IOException && e.Message.Contains("Unable to read data from the transport connection: The I/O operation has been aborted because of either a thread exit or an application request");
+
         [Fact]
+        [RetryTest(nameof(ConnectionNotClosedWhenClientSatisfiesMinimumDataRateGivenLargeResponseHeadersRetryPredicate),
+            "Active investigation into potential corefx sockets bug: https://github.com/dotnet/corefx/issues/30691",
+            OperatingSystems.Windows,
+            5)]
```

Comment:
I'm increasing the retry limit here since when the tests fail, they fail quickly and it may take up to 3 runs total to pass. I'm setting it to 5 which should be a safer limit.

This is beginning to seem like a bad state somewhere not being cleared promptly.